builtin/providers/aws/resource_aws_api_gateway_api_key.go

```diff
@@ -0,0 +1,225 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsApiGatewayApiKey() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsApiGatewayApiKeyCreate,
+		Read:   resourceAwsApiGatewayApiKeyRead,
+		Update: resourceAwsApiGatewayApiKeyUpdate,
+		Delete: resourceAwsApiGatewayApiKeyDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"enabled": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"stage_key": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"api_id": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"stage_name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsApiGatewayStageKeys(d *schema.ResourceData) []*apigateway.StageKey {
```

Comment:
Can you move this whole function into `builtin/providers/aws/structure.go`, please? That's typically where we keep all similar functionality.